src/index.js

```diff
@@ -164,9 +164,11 @@ class ServerlessAppSyncSimulator {
    * Resolves resourses through `Ref:` or `Fn:GetAtt`
    */
   resolveResources(toBeResolved) {
-    // Pass Resources to allow Fn::GetAtt resolution
+    // Pass all resources to allow Fn::GetAtt and Conditions resolution
     const node = {
-      Resources: this.serverless.service.resources?.Resources || {},
+      // Conditions must appear before Resources, see https://github.com/bboure/serverless-appsync-simulator/pull/31#issuecomment-633872077
+      Conditions: this.serverless.service.resources?.Conditions || {},
+      ...this.serverless.service.resources,
```

Comment:
@JReinhold is there anything else than `Resources` and `Conditions` that could come in `this.serverless.service.resources`  ?

Reply:
As per the docs, yes: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-anatomy.html

(Resources, Conditions, Transform, Outputs, Mappings, Parameters, maybe Metadata, and Description)
I don't know if all of them are relevant in Serverless Framework, or even for this appsync plugin. I know that we use Outputs in our projects, so that is at least one.

I don't know of a scenario where Outputs will break anything, but here I just wanted to be on the safe side and include everything, especially because the error is very hard to debug, so it's hard to figure out that the issue is with missing resources.

Reply:
Yeah, got you. `Outputs` should definitely be included. 

Now, we should probably open an issue on `cfn-resolver-lib` and see how we can resolve this. I am wondering if this is the right solution here.

Why?
Any Node could reference any other.

Example: 
`Conditions` could also reference `Resources`, and then we end up with the opposite problem.

````yml
Conditions:
  SomeCondition:
    Fn::Equals: [ "Fn::GetAtt": ["SomeResource", "Name"], "something"]
Resources:
 SomeResource:
   # ...
````
Here, `Resources` should be resolved first.

Same goes for `Outputs`, etc.

Nodes should probably be resolved Lazily since we cannot know the right order before hand.

I would rollback this, and come back to your original solution, then open an issue in `cfn-resolver-lib`.

Since each use case is different, and before we solve this in the resolver, each user should probably make sure that the Resources/Conditions/etc are sorted in the right order in the yml file...

Reply:
Yeah, that's a good point. By not forcing any order, we can let the user handle the ordering, as in my example where I defined the Conditions in all my files to make sure they were on top.

Reply:
Awesome.

Thank you